packages/backend/src/models/SpaceModel.ts

```diff
@@ -1383,7 +1384,9 @@ export class SpaceModel {
                     is_private: isPrivate,
                     name,
                     created_by_user_id: userId,
-                    slug: await generateUniqueSlug(trx, SpaceTableName, slug),
+                    slug: forceSameSlug
```

Comment:
generateUniqueSlug doesn't check if the slug are unique within a project

On promotion, sometimes we create spaces when promoting, this means, the new space in the upstream project had  adifferent slug. We need to force the slug to be the same, otherwise promotion is useless. 

Another option is to have ` generateUniqueSlug` to search for unique slugs within a project, and throw errors on the service if they are not the same, but this seems a bit overcomplicated.